### PR TITLE
convertTouchMoveStartToButtonDownAndMotion stopped working with change to not await snap.

### PR DESCRIPTION
**Pull request by bbastings**
Tools that want to turn onTouchMoveStart into onMouseStartDrag are no longer working.
  
public override async onTouchMoveStart(ev: BeTouchEvent, startEv: BeTouchEvent): Promise<EventHandled> {
   return IModelApp.toolAdmin.convertTouchMoveStartToButtonDownAndMotion(startEv, ev);
}

The view clip modify arrow controls among other things rely on this. When we are explicitly trying to start a drag operation, we need to make sure we await  checking for a snap in onMotion in order for the active tool to see if it wants to handle the onTouchMoveStart event before allowing it to be passed on to the IdleTool.

Not sure if there's a cleaner way to do this, just adding Promise.resolve when forcing a start drag was very bad.

### CODE REVIEW

File: core/frontend/src/tools/ToolAdmin.ts [hunk 1]
```diff
@@ -1008,15 +1003,31 @@ export class ToolAdmin {
         if (undefined !== tool && isValidLocation)
           tool.receivedDownEvent = true;
 
-        return this.onStartDrag(ev, isValidLocation ? tool : undefined);
+        await this.onStartDrag(ev, isValidLocation ? tool : undefined);
+        return;
```
Comment (pmconne) [inline core/frontend/src/tools/ToolAdmin.ts:1007]: superfluous `return`.

**Review by pmconne — APPROVED** [review]

File: core/frontend/src/tools/ToolAdmin.ts (hunk 1 above)
Comment (bbastings) [reply core/frontend/src/tools/ToolAdmin.ts:1007]: Keith changed that...he preferred not return EventHandled when no one checks the status...

Comment (kabentley) [reply core/frontend/src/tools/ToolAdmin.ts:1007]: no it's not.

**Review by kabentley — APPROVED** [review]

File: core/frontend/src/tools/ToolAdmin.ts (hunk 1 above)
Comment (pmconne) [reply core/frontend/src/tools/ToolAdmin.ts:1007]: I see -  you changed it after I read it.

Comment (kabentley) [reply core/frontend/src/tools/ToolAdmin.ts:1007]: haha - you were commenting about the `return` that i removed, but your comment happened to fall on another `return` after my changes.

**Review by tcobbs-bentley — APPROVED** [review]

**Comment by calebmshafer** [comment]
@Mergifyio update

**Pull request merged**